Source/WebCore/dom/NodeInlines.h

```diff
@@ -244,6 +245,12 @@ inline ContainerNode* Node::parentNodeGuaranteedHostFree() const
     return parentNode();
 }
 
+inline void Node::setTreeScope(TreeScope& scope)
+{
+    Locker locker { TreeScope::treeScopeMutationLock() };
+    m_treeScope = &scope;
```

Comment:
I can't put `WTF_GUARDED_BY_LOCK` on `m_treeScope` because the main thread need to access it without grabbing a lock.